python/lib/dcos/dcos/util.py

```diff
@@ -280,7 +280,7 @@ def read_file_secure(path):
     enforce_file_permissions(path)
 
     with open_file(path) as file_:
-        return file_.read()
+        return file_.read().rstrip()
```

Comment:
Perhaps only strip newline-characters, e.g. `.rstrip('\r\n')` will not strip tabs or spaces, which may be valid password characters?

Reply:
That's what I had in the beginning then followed @jgehrcke's suggestion, I think both are fine.

If users are bit by this I'd be interested to know why they think putting passwords with trailing spaces / tabs in files and expecting them to be processed successfully by other users / tools is a good idea :confused: